packaging: sync spec file with Fedora package

praiskup opened the pull request:
Fedora maintainers have done big cleanup and simplification so it
is worth to sync the spec file here.

Also, as it is now suggested to build from github release tarball,
which is not 100% perfect, this commit fixes the 'srpmgen' tool so
it is able to generate equivalent tarball directly from git repo.

vlsi commented:
Just in case: https://travis-ci.org/pgjdbc/pgjdbc/jobs/153235670

copr build failed trying to download `http://pgjdbc-rel9.5.git.tar.gz/`

```
 * getting sources pgjdbc-REL9.5.git.tar.gz
--2016-08-18 11:02:23--  http://pgjdbc-rel9.5.git.tar.gz/
Resolving pgjdbc-rel9.5.git.tar.gz (pgjdbc-rel9.5.git.tar.gz)... failed: Name or service not known.
wget: unable to resolve host address 'pgjdbc-rel9.5.git.tar.gz'
can't wget pgjdbc-REL9.5.git.tar.gz at /usr/bin/srpmgen line 64, <$config_file> line 10.
The command "./.travis/travis_build.sh" exited with 2.
```

praiskup commented:
Hi @vlsi, as I changed some scripts, I've changed the docker image too (and I probably forgot
to push the image to dockerhub).  Can you re-submit the testsuite now please?

vlsi commented:
Thanks for looking into this.
Here's Travis link: https://travis-ci.org/pgjdbc/pgjdbc/builds/153495206
There's a queue for pgjdbc, so fedora_build branch might wait like an hour in the build queue.

vlsi commented:
@praiskup ,

As far as I can understand, the problem is
```
source0:
  git_archive:
    prefix: pgjdbc-REL9.5.git
    tarball_base: REL9.5.git
```

It basically makes git archive into `REL9.5.git.tar.gz`, and later tries to check `pgjdbc-REL9.5.git.tar.gz` file for accessibility. That file is obviously not there, so it tries to wget the file and it fails.

I think here's the fix: https://github.com/pgjdbc/pgjdbc/commit/742622d99e2202872d5dbde4c7bdb686f06b5290, however it looks like you need to integrate that into the docker image so it would be available to Travis.

vlsi commented:
@praiskup , I've noticed that "license" is marked as "BSD and ASL 2.0"

https://github.com/pgjdbc/pgjdbc/blob/master/packaging/rpm/postgresql-jdbc.spec.tpl#L50-L51

I don't think we have Apache 2.0 bits in the final build.
We do use ASL2 tools during compile time, however we don't ship ASL2 bits.

The commen "ASL 2.0 applies only to postgresql-jdbc.pom file" looks weird.

vlsi commented:
@praiskup , just in case: copr build still fails.

praiskup commented:
@vlsi sorry for long delay, I'll have a look asap

praiskup commented:
Good catch with the license, I'll sync this PR with:
http://pkgs.fedoraproject.org/cgit/rpms/postgresql-jdbc.git/commit/?id=9b77a0e3ea2b09b5d577

praiskup commented:
@vlsi, the issue with tarball was that I haven't updated the `Source` tag in spec file according
to the format provided  by github, iow: I used `psqlodbc-REL*` instead of `REL*`.

vlsi commented:
Oh, it seems to pass somehow: https://travis-ci.org/pgjdbc/pgjdbc/builds/155903857

praiskup commented:
:) right, that's expected after `psqlodbc-REL*` to `REL*` change.
